File: Chapter03/3.3.1 SRP/BadWay/Player.cs
namespace Srp.BadWay
{
    public class Player
    {
        public string Name { get; set; }

        // Responsibility 1: Player’s own state/abilities
        public void DribbleBall() { Console.WriteLine("Dribbling ball..."); }

        // Responsibility 2: Tactical Logic
        public void DetermineBestPosition()
        {
            Console.WriteLine("Determining best position...");
        }

        // Responsibility 3: Data Persistence
        public void SaveStatsToDatabase()
        {
            Console.WriteLine("Saving stats to database...");
        }
    }
}

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.3.1 SRP/After/Player.java
package com.grokkingsoftwarearchitecture.chapter03.srp.after;

// Responsibility 1: Manages only the player’s state and actions
public class Player {
    public String name;

    public Player(String name) {
        this.name = name;
    }

    public void dribbleBall() {
        System.out.println("  [Action] " + name + " is dribbling the ball down the court.");
    }
}